test_suite/package/test_otel.py

```diff
@@ -0,0 +1,149 @@
+        cmd_output = json.loads(command_output)
+
+        instance_status = cmd_output["InstanceStatuses"][0]["InstanceStatus"]["Status"]
+        system_status = cmd_output["InstanceStatuses"][0]["SystemStatus"]["Status"]
+
+        if instance_status == "ok" and system_status == "ok":
+            break
+        else:
+            logger.info("Instance status is not 'passed' yet. Waiting...")
+            time.sleep(5)
+
+
+@pytest.fixture()
+def modify_iam_role(instance_data, host):
+    instance_id = instance_data['instance_id']
+    region = instance_data['availability_zone'][:-1]
+    iam_role_name = "CloudWatchAgentServerRole_2"
+
+    command_to_run = [
+        'aws', 'ec2', 'associate-iam-instance-profile',
+        '--instance-id', instance_id,
+        '--region', region,
+        '--iam-instance-profile', 'Name="{}"'.format(iam_role_name)
+    ]
+    modify_iam_role_cmd = ' '.join(command_to_run)
+
+    assert host.backend.run_local(modify_iam_role_cmd), 'faild to update iam role'
+
+
+@pytest.fixture()
+def install_packages(request, host):
+    class_instance = request.node.cls
+    install_cmd = f'yum install -y {class_instance.package_name} --nogpgcheck'
+    repo_path = ('https://copr.fedorainfracloud.org/coprs/miyunari'
+                 '/redhat-opentelemetry-collector/repo/rhel-9/miyunari-redhat-opentelemetry-collector-rhel-9.repo')
+    with host.sudo():
+        assert host.run(f'yum-config-manager --add-repo {repo_path}').succeeded
+        assert host.run(install_cmd).succeeded, f'Failed to install the package {class_instance.package_name}'
+        test_lib.print_host_command_output(host, "rpm -qa | grep opentelemetry*")
+
+
+@pytest.fixture()
+def start_service(request, host):
+    class_instance = request.node.cls
+    start_enable_service = (
+        f'systemctl start {class_instance.package_name} && '
+        f'systemctl enable {class_instance.package_name}'
+    )
+
+    with host.sudo():
+        assert host.run(start_enable_service).succeeded, (f'Failed to start the service {class_instance.package_name}')
+        assert host.service(class_instance.package_name).is_enabled, (
+            f'Failed to enable the service {class_instance.package_name}'
+        )
+        assert host.service(class_instance.package_name).is_running, (
+            f'Failed to run the service {class_instance.package_name}'
+        )
+
+    def finalizer():
+        console_lib.print_divider(f'Removing the package {class_instance.package_name}')
+        assert host.run(f'sudo yum remove -y {class_instance.package_name}')
+        assert not host.check_output(f'rpm -q {class_instance.package_name}')
+        assert not host.run(f'ssh {class_instance.instance_dns}').succeeded
+        logger.info("Verify logs don't appear")
+        # TODO: verify logs don't appear.
+    request.addfinalizer(finalizer)
+
+
+@pytest.fixture(scope="function", autouse=True)
```

Comment:
I am still iffy about autousing this fixture at the function level. Auto-registration is time-consuming.

Today this file has only one test case. However, _if by any chance_ we add more test cases and forget about this, it will take too much time to run, and it may also fail since the system will already be subscribed. Let's make it run only once.

Reply:
No problem. changing the scope to "session", this ensures that the fixture is executed only once, regardless of how many test functions or classes are using it throughout the entire test session.